server/utils/notifications/dispatcher.py

```
@@ -139,37 +141,74 @@ def _enrich_incident_summary(incident_data: Dict[str, Any], session_id: str, use
+                    started = started.astimezone(timezone.utc)
+                return row[0], started
+    except Exception as e:
+        logger.warning("[Dispatcher] Failed to fetch action run details: %s", e)
+    return _UNKNOWN_ACTION, None
+
+
+def _fetch_last_bot_message(session_id: str, user_id: str) -> Optional[str]:
+    """Fetch the last bot message text from a chat session."""
+    try:
+        with db_pool.get_admin_connection() as conn:
+            with conn.cursor() as cur:
```

Ruff (0.15.17)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

Combine `with` statements

(SIM117)